contracts/Gauge.sol

```diff
@@ -334,63 +507,55 @@ contract Gauge is IGauge {
+                IExtraReward(extraRewards[i]).getRewardFor(msg.sender);
             }
         }
     }
 
-    function donate(uint256 _amount) external returns (bool) {
-        require(_amount != 0);
-        IERC20(rewardToken).safeTransferFrom(
-            msg.sender,
-            address(this),
-            _amount
-        );
-        queuedRewards = queuedRewards + _amount;
+    /**
+     * @notice
+     * Transfer penalty to the veYFIRewardContract
+     * @dev Penalty are queued in this contract.
+     * @return true
+     */
+    function transferQueuedPenalty() external returns (bool) {
+        uint256 toTransfer = queuedPenalty;
+        queuedPenalty = 0;
+
+        IERC20(rewardToken).approve(veYfiRewardPool, toTransfer);
+        BaseGauge(veYfiRewardPool).queueNewRewards(toTransfer);
+        emit TransferedQueuedPenalty(toTransfer);
         return true;
     }
 
-    function queueNewRewards(uint256 _rewards) external returns (bool) {
-        require(_rewards != 0);
-        IERC20(rewardToken).safeTransferFrom(
-            msg.sender,
-            address(this),
-            _rewards
+    /**
+     * @notice
+     * set reward manager
+     * @dev Can be called by rewardManager or owner
+     * @param _rewardManager new reward manager
+     * @return true
+     */
+    function setRewardManager(address _rewardManager) external returns (bool) {
+        require(
+            msg.sender == rewardManager || msg.sender == owner(),
+            "!authorized"
         );
-        _rewards = _rewards + queuedRewards;
 
-        _notifyRewardAmount(_rewards);
-        queuedRewards = 0;
+        require(_rewardManager != address(0), "_rewardManager 0x0 address");
+        rewardManager = _rewardManager;
+        emit UpdatedRewardManager(rewardManager);
         return true;
     }
 
-    function _notifyRewardAmount(uint256 reward)
+    function _notProtectedTokens(address _token)
         internal
-        updateReward(address(0))
+        view
+        override
+        returns (bool)
     {
-        historicalRewards = historicalRewards + reward;
-        if (block.timestamp >= periodFinish) {
-            rewardRate = reward / DURATION;
-        } else {
-            uint256 remaining = periodFinish - block.timestamp;
-            uint256 leftover = remaining * rewardRate;
-            reward = reward + leftover;
-            rewardRate = reward / DURATION;
-        }
-        currentRewards = reward;
-        lastUpdateTime = block.timestamp;
-        periodFinish = block.timestamp + DURATION;
-        emit RewardAdded(reward);
-    }
-
-    function transferQueuedPenalty() public {
-        uint256 toTransfer = queuedPenalty;
-        queuedPenalty = 0;
-
-        IERC20(rewardToken).safeApprove(veYfiRewardPool, toTransfer);
-        IVeYfiRewardPool(veYfiRewardPool).donate(toTransfer);
+        return
+            _token != address(rewardToken) && _token != address(stakingToken);
```

Comment:
VeYfiRewards.sol also checks IVotingEscrow(veToken).migration(), do we need to do that here?